ratelimit.go

```diff
@@ -0,0 +1,87 @@
+package strava
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type RateLimit struct {
+	NextRequestTime time.Time
+	LimitShort      int
+	LimitLong       int
+	UsageShort      int
+	UsageLong       int
+}
+
+var RateLimiting RateLimit
+
+func (ratelimit *RateLimit) Exceeded() bool {
+	if time.Now().After(ratelimit.NextRequestTime) {
+		return false
+	} else {
+		return true
+	}
+}
+
+func (ratelimit *RateLimit) FractionReached() float32 {
```

Comment:
float32 or float64, the eternal question :)  The rest of the library uses float64 everywhere else.